influxdb3_write/src/write_buffer/table_buffer.rs

```diff
@@ -64,16 +68,22 @@ impl TableBuffer {
                         }
                     }
                     FieldData::Tag(v) => {
-                        let b = self.data.entry(f.name).or_insert_with(|| {
+                        if !self.data.contains_key(&f.name) {
                             let mut tag_builder = StringDictionaryBuilder::new();
                             // append nulls for all previous rows
                             for _ in 0..(row_index + self.row_count) {
                                 tag_builder.append_null();
                             }
-                            Builder::Tag(tag_builder)
-                        });
+                            self.data.insert(f.name.clone(), Builder::Tag(tag_builder));
+                        }
+                        let b = self
+                            .data
+                            .get_mut(&f.name)
+                            .expect("tag builder should exist");
```

Comment:
This could simplified to this I believe:

```rust
let b = self.data.entry(&f.name)
    .or_insert_with(|| {
        let mut tag_builder = StringDictionaryBuilder::new();
        // append nulls for all previous rows
        for _ in 0..(row_index + self.row_count) {
            tag_builder.append_null();
        }
        self.data.insert(f.name.clone(), Builder::Tag(tag_builder));
    });
```

Reply:
entry API won't take a borrow

Reply:
ah dang. That's unfortunate.

Reply:
One day we might want to consider this option, but it feels very arm twisting to get the compiler to accept this:

https://idubrov.name/rust/2018/06/01/tricking-the-hashmap.html